app/routes/questionnaire.py

```diff
@@ -77,13 +77,14 @@ def before_questionnaire_request():
 
 @post_submission_blueprint.before_request
 @login_required
-def before_post_submission_request():
+@with_questionnaire_store
+def before_post_submission_request(questionnaire_store):
     if request.method == "OPTIONS":
         return None
-
     session_store = get_session_store()
     session_data = session_store.session_data
-    if not session_data.submitted_time:
+
+    if not questionnaire_store.submitted_at:
```

Comment:
The decorator shouldn't be used here, it needs to be fetched using `get_questionnaire_store` after the `OPTIONS` check. If you make an `OPTIONS` requests without a session, this will generate a 500.

Also, the call to `get_session_store()` should be moved below the `submitted_at` check and only called when needed.